datapipelines/src/main/kotlin/io/customer/sdk/CustomerIO.kt

```diff
@@ -224,9 +228,9 @@ class CustomerIO private constructor(
      * Common method to track an event with traits.
      * All other track methods should call this method to ensure consistency.
      */
-    override fun <T> track(name: String, properties: T, serializationStrategy: SerializationStrategy<T>) {
+    override fun <T> track(name: String, properties: T, serializationStrategy: SerializationStrategy<T>, enrichment: EnrichmentClosure?) {
         logger.debug("track an event with name $name and attributes $properties")
-        analytics.track(name = name, properties = properties, serializationStrategy = serializationStrategy)
+        analytics.track(name = name, properties = properties, serializationStrategy = serializationStrategy, enrichment = enrichment)
     }
```

Comment:
I think we could have avoided exposing `EnrichmentClosure` to public for now by adding private functions like following

```kotlin
private fun <T> track(name: String, properties: T, serializationStrategy: SerializationStrategy<T>, enrichment: EnrichmentClosure?)
private fun fun deleteDeviceToken(enrichment: EnrichmentClosure?)
```

And calling them with `null` or desired values from current `public` methods.

So unless we really want them to be exposed to public or wrapper SDKs, I'm okay with this. But if not, may be we can try not changing public APIs for this?

And if we do so, rest of the suggestions can be ignored as they will no longer be needed if we don't change any public apis.

Reply:
i think it would be a suited solution for now.